substrate/frame/support/src/traits/tokens/imbalance/imbalance_accounting.rs

```diff
@@ -0,0 +1,44 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Convenience trait for working with dynamic type of Imbalance.
+
+use alloc::boxed::Box;
+
+/// Helper trait to be used for generic Imbalance, helpful for tracking multiple concrete types of
+/// `Imbalance` using dynamic dispatch of this trait.
+pub trait ImbalanceAccounting<Balance> {
```

Comment:
this could be `ImbalanceAccounting: Unsafe`
where we decouple `subsume_amount`, `unsafe_clone` and `forget_amount` to `Unsafe`

Reply:
done